system/metrics/sinks/elastic/elasticSink.cpp

```diff
@@ -41,67 +43,136 @@ extern "C" MetricSink* getSinkInstance(const char *name, const IPropertyTree *pS
+    // Validate the host configuration minimal settings are present
+    if (hostDomain.isEmpty() || hostProtocol.isEmpty())
     {
-        elasticHostUrl.append(hostProtocol).append("://").append(hostName).append(":").append(hostPort);
+        WARNLOG("ElasticMetricSink: Host configuration missing domain and/or protocol");
+        return false;
+    }
+
+    // build url for use with httplib Client
+    elasticHostUrl.append(hostProtocol).append("://").append(hostDomain);
+    if (!hostPort.isEmpty())
+        elasticHostUrl.append(":").append(hostPort);
+
+    // Read optional certificate file path
+    pHostConfigTree->getProp("@certificateFilePath", certificateFilePath);
+
+    // Get authentication settings, if present
+    Owned<IPropertyTree> pAuthConfigTree = pSettingsTree->getPropTree("authentication");
+    if (!pAuthConfigTree)
+        return true;
+
+    // Retrieve the authentication type and validate (only basic is supported)
+    if (!pAuthConfigTree->getProp("@type", authenticationType) || !streq(authenticationType, "basic"))
+    {
+        WARNLOG("ElasticMetricSink: Only basic authentication is supported");
+        return false;
+    }
+
+    StringBuffer credentialsSecretKey;
+    pAuthConfigTree->getProp("@credentialsSecret", credentialsSecretKey);  // vault/secrets key
+    if (!credentialsSecretKey.isEmpty())
+    {
+        StringBuffer credentialsVaultId;
+        pAuthConfigTree->getProp("@credentialsVaultId", credentialsVaultId);//optional HashiCorp vault ID
+
+        PROGLOG("Retrieving ElasticSearch host authentication username/password from secrets tree '%s', from vault '%s'",
+               credentialsSecretKey.str(), !credentialsVaultId.isEmpty() ? credentialsVaultId.str() : "");
+
+        Owned<const IPropertyTree> secretTree(getSecret("authn", credentialsSecretKey.str(), credentialsVaultId, nullptr));
+        if (secretTree == nullptr)
+        {
+            WARNLOG("ElasticMetricSink: Unable to load secret tree '%s', from vault '%s'", credentialsSecretKey.str(),
+                    !credentialsVaultId.isEmpty() ? credentialsVaultId.str() : "n/a");
+            return false;
+        }
+
+        // authentication type defines the secret key name/value pairs to retrieve
+        if (streq(authenticationType, "basic"))
+        {
+            if (!getSecretKeyValue(username, secretTree, "username") || !getSecretKeyValue(password, secretTree, "password"))
+            {
+                WARNLOG("ElasticMetricSink: Missing username and/or password from secrets tree '%s', vault '%s'",
+                        credentialsSecretKey.str(), !credentialsVaultId.isEmpty() ? credentialsVaultId.str() : "n/a");
+                return false;
+            }
+        }
     }
     else
     {
-        WARNLOG("ElasticMetricSink: Host configuration missing or invalid");
+        // if basic auth, username and password are stored directly in the configuration
```

Comment:
should require configuration via a secret

Reply:
If a secret is required for the username and password, how would that work in a bare metal environment? Is it possible to load secrets for bare metal?